otherlibs/stdune/src/option.mli

```diff
@@ -60,3 +60,19 @@ module List : sig
 end
 
 val merge : 'a t -> 'a t -> f:('a -> 'a -> 'a) -> 'a t
+
+module Unboxed : sig
+  (** Poor man's unboxed option types *)
```

Comment:
Surely, you can't use this type for any `'a`? Can you document the conditions under which it's safe?

Reply:
(Like, I'm guessing `'a` can't be an immediate?)

Reply:
I've documented this better and improved the errors.